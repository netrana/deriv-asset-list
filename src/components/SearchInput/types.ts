import { InputProps } from "antd";

export interface Props extends InputProps {
  onSearch: (value: string) => void;
}
