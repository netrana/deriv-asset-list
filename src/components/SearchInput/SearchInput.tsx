import React, { FC } from 'react';
import { Input, Space } from 'antd';

import {Props} from './types';

const { Search } = Input;


const SearchInput: FC<Props> = (props) => {

  const { className = '', onSearch  } = props

  return (
    <Space direction="vertical">
      <Search
        placeholder="input search text"
        onSearch={onSearch}
        style={{ width: 200, padding: '0 1rem .5rem 0' }}
        allowClear
      />
    </Space>
  );
};
export default SearchInput;
