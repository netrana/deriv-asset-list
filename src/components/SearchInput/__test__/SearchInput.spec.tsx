import React from "react";
import render from "react-test-renderer";

import SearchInput from "../SearchInput";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

jest.mock("antd", () => ({
  Input: {
    Search: "Search",
  },
  Space: "Space",
}));

describe("SearchInput component test", () => {
  describe("snapshot ", () => {
    it("should render correctly", () => {
      const tree = render.create(<SearchInput onSearch={jest.fn} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
