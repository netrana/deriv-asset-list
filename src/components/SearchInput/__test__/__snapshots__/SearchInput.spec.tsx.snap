// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component test snapshot  should render correctly 1`] = `
<Space
  direction="vertical"
>
  <Search
    allowClear={true}
    onSearch={[Function]}
    placeholder="input search text"
    style={
      Object {
        "padding": "0 1rem .5rem 0",
        "width": 200,
      }
    }
  />
</Space>
`;
